scipy/optimize/minpack.py

```diff
@@ -671,6 +672,20 @@ def curve_fit(f, xdata, ydata, p0=None, sigma=None, absolute_sigma=False,
     >>> plt.show()
 
     """
+
+    def check_len(z):
+        if type(z) == int or type(z) == float:  
+            return False
+        else:
+            z = np.array(z)
+            if z.any() == np.inf or type(z.any()) == float:
+                return False
+            else:
+                return len(z) != 0
+
+    if check_len(xdata) is False or check_len(ydata) is False:
```

Comment:
try never to write `is False`, that's not very idiomatic Python. Here better: `if not check_len(xdata) or not check_len(ydata):`

Reply:
More importantly, what are you trying to achieve with `check_len`? Checking for empty arrays can be done like so:

    if xdata.size == 0 or ydata.size == 0:

Also, `TypeError` is not the right thing to raise, should be `ValueError`.

Reply:
I will change `is False` but while using size we get an error saying that `AttributeError: 'list' object has no attribute 'size'`

Reply:
I changed the in to `ValueError`